big-int/src/operators/equal_to.cpp

```diff
@@ -44,11 +44,66 @@ bool largeInt::operator==(const largeInt &z)
     }
     for (int i = 0; i < this->number.length(); i++)
     {
+        // iterates through all digits of both inputs
+        // if two digits are not the same, returns false
         if (this->number[i] != z.number[i])
         {
             return false;
         }
     }
     return true;
 }
-} // namespace libbig
+
+bool largeInt::operator==(int z)
+{
+    // checks if both inputs have the same sign (positive or negative)
+    if (this->sign != (z >= 0))
+    {
+        return false;
+    }
+    // iterates through each digit, starting by the last
+    // if two digits are not the same, returns false
+    if (z < 0)
+    {
+        z *= -1;
+    }
+    int n = this->number.length() - 1;
+    while (z != 0)
+    {
+        if (this->number[n] != z % 10)
```

Comment:
this must be
`this->number[n] - '0' != z % 10)`
else it will end up comparing a char with int

Reply:
fixed